### DISCUSSION TITLE
SolaX Inverter / EVC Automations

### DISCUSSION THREAD

**Discussion by wranglatang** [wills106/homeassistant-solax-modbus#919] [General]
Has anyone got any example automations that they use with their SolaX EV Charger to do scheduling during off peak times or day charging when there is surplus energy in the day?

**Comment by PatrikTrestik** [comment]
Here is my setup:
* most of the time I use Green mode and charge EV from PV.
* I have automation to monitor EV SOC (provided by car OVMS) and stop charging on set value. 80% most of the time to save car battery.
* I have automation to start Fast charging if price of electricity drops below set limit (ussually set to 0). I use it to charge when there is negative price.

I don't schedule charging. I created calendar schedule in HA to reflect low price and I calculate price for each hour. I drive automations based on this price sensor.

**Reply by wranglatang** [reply]
Okay! That makes sense! At the moment, due to the issue with my car charging from the SolaX batteries, if i use green mode - as soon as i plug in it start charging. What i have currently had to setup is some Restrict Time the EVC and then i just leave it on fast. I have a cheaper rate at night, when this starts Home Assistant sets my Inverter to "Manual" so it doesnt charge or discharge.

Really want to get green mode working!

**Reply by PatrikTrestik** [reply]
I would suggest to solve Green mode first. If there is working communication betveen EVC and Inverter you can get what you want with less automations.

What do you see in EVC settings for Grid Data Source?
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/17616747/4d543275-54b4-4508-9e2e-f1e52cc24fb7)

**Reply by wranglatang** [reply]
Hmmm. I don’t have that option. This is what I see under Settings -> Advanced Settings:
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/30660751/fb7eea43-95d7-4ab5-b5cb-774100e696a6)

**Reply by wranglatang** [reply]
It’s interesting actually. As if i look at the HTTP API i see this in the diagnostic section: 
<img width="294" alt="image" src="https://github.com/wills106/homeassistant-solax-modbus/assets/30660751/d89095a1-3be6-473b-8da3-6f52fbedb149">

But if i try to change the meter setting to “inverter” it just defaults back to CT!! Must be a F/W issue right!?

**Reply by PatrikTrestik** [reply]
It is normal and expected behaviour. Indicates that your inverter is not connected.

Contact installer and let them check rs485 wiring from EVC to Inverter.

Without communication you smart functions will not work.

**Reply by wranglatang** [reply]
Okay. I will double check that next week. It’s odd as the readings for PV Power are working.

**Reply by PatrikTrestik** [reply]
Your PV Power is most probably showing only "excess/exportted" PV. But if correctly connected it should be PV+Battery charge.
Check if you see PV Power when your batter is charging. Charging power should be included.

**Reply by wranglatang** [reply]
Okay! So i think things are now working! When the EVC is in green mode and i plug the car in, it starts to charge for about a minute then switches off. It appears that when i put the EVC into fast mode it still charges from battery however.

**Reply by Philipp-X1** [reply]
If there is not enough PV surplus it will stop in around a minute. This is correct behaviour. It should start again when there is enough surplus PV Power.

**Comment by lieglw** [comment]
Hello, does anybody know how to change the charge current. Both are not working for me, so the number.bmw_datahub_charge_current (as I do not have a data hub) and number.bmw_charge_current. I have tried and it shows that it was checked, but when I was looking to the max charge current setting of the solax HTTP API it staid untouched.
I have unlocked the charger and also used the EV Sync RTC. Also by starting it does not show an effect on this.
Am I doing something wrong or is.
Is there a possibility to send surplus information? Then the Green Mode can be used, perhaps.

I use at the moment node red and it starts to work 😂, but I am using the HTTP API and control the EVC by starting charging by "Fast" and depending on the surplus calculation adjusting the  max charge current setting. 
But of course by using the Modbus there are other possibilities.

**Reply by Philipp-X1** [reply]
If you want to control the EVC you have to Set it to Mode Fast and change Max charge current. This should work. 

If you are interested in simulating Green Mode you should take a look at Patriks work here:
https://github.com/PatrikTrestik/homeassistant-solax-evc-proxy/discussions/1

**Reply by lieglw** [reply]
Hi, in the Modbus version there is no set max charge current like in the http API version. So I do not find it, I also checked the hidden entities. 
I have tried this with the number.*_charge_current but this had no effect on the max current setting. So it ramps up to max by using fast.

**Reply by lieglw** [reply]
By the way changing the Green Gear works.
So could it be that the max current set is missing somehow.

**Reply by PatrikTrestik** [reply]
You are all right. I received latest Modbus description (somewhere here in other discussion) and there are promissing registers which should allow charge current regulation. 
Tomorrow I have time to check and test.

If I found something usefull, I'll add sensor to modbus integration.

**Reply by PatrikTrestik** [reply]
Modbus register 0x628 is right one. It allows current regulation with precision 0.01A.
It is intended for temporary custom regulation. It is not stored after power loos or restart.
Tested. My measured current was cca 0.5A offset, but it changed correctly during Fast charge.
6-32A(16A).

There is register 0x668 which is also 0.01A and its purpouse is for limiting all modes power. It is stored after restart so it might be stored in EEPROM. But this is accesible in SolaX app as Maximum Charge Current setting.

**Reply by lieglw** [reply]
@PatrikTrestik Thank You, I use directly the register 0x628 in Node Red and it works perfectly.
For start and stop and this stuff I use your http integration as it sets Run Mode and Use Mode correct and I have not to bother.
Writing to the Charger Mode register did not work for me and I found no possibility to have this functionality with Modbus. It seems that then run and use mode are not set automatically (e.g changing from available to charging when connecting the car)

**Comment by Philipp-X1** [comment]
You are right. It is not Max charge current. It is number.*_charge_current. It should work. 
I am using this aswell, see here:

![IMG_7346](https://github.com/user-attachments/assets/a872640f-fc5e-48a9-bca7-ec6192832522)

It does not affect Max charge current. I think Max charge current is only used to prevent overload.
